### PR TITLE
Stub out tzdb support to unblock `Now` and `ZonedDateTime`

### PR DESCRIPTION
**Pull request by nekevss**
And thus begins the Great Time Zone / `ZonedDateTime` implementation. :sweat_smile: 

This stubs out `tzdb` support by relying solely on the user's zoneinfo files, and should allow us to move forward with implementing and testing `ZonedDateTime` without being blocked by a proper tzdb provider for Boa.

This PR implements:

  - `Now` builtin
  - `FsTzdbProvider` stub
  - Some `ZonedDateTime` APIs along with experimental, non-engine focused APIs.
  
Things of note that are not in this PR:
  
  - Handling POSIX's [proleptic tz format](https://snapshots.sourceware.org/glibc/trunk/2024-06-27_16-55_1719507301/manual/html_node/Proleptic-TZ.html) and non-fat version of the TZDB.
  - Any thing zoneinfo / `zic` related.
  - Tests for more time zone edge cases
  - Many `ZonedDateTime` methods and abstract ops
  - **Windows Support**
  
Let me know what you think :)

### CODE REVIEW

**Review by jasonwilliams — COMMENTED** [review]
LGTM! 
Good start, nice to have a way forward which will unblock us for now

File: Cargo.toml
Comment (nekevss) [reply Cargo.toml:32]: Good catch! Forgot to remove it when I moved to `std`.

File: src/tzdb.rs
Comment (nekevss) [reply src/tzdb.rs:45]: No, there's not really a windows equivalent, so we will need to package `tzdb` for windows.

That's where we may be able to use `jiff_tzdb` as an intermediate step, but we would need to add steps for resolving from the POSIX tz string as jiff is compiled in the "slim" format of `tzdb` while linux and macos are compiled in "fat"

File: src/tzdb.rs
Comment (jedel1043) [inline src/tzdb.rs:136]: It took me a bit of time to understand that this is just checking if the sum of offsets is positive, so maybe just do that?
```suggestion
let shift_window = usize::from(record.utoff + record_minus_one.utoff >= Seconds(0))
```

File: src/tzdb.rs
Comment (jedel1043) [inline src/tzdb.rs:118]: Can't we just return the timezone when the binary search finds a corresponding entry (`Ok(idx)`)?

File: src/tzdb.rs
Comment (jedel1043) [inline src/tzdb.rs:133]: I don't think this handles indexing to the transition table correctly. What if the estimated index is one past the end of the table, because the epoch is bigger than any of the transition times?

File: src/tzdb.rs
Comment (jedel1043) [inline src/tzdb.rs:103]: Can you document what this function does? I _think_ it's just trying to determine the correct timezone data from the current epoch in local time, but it would be good to have it documented.

**Review by jedel1043 — COMMENTED** [review]
Thank you for all your work! I'm really excited to start playing with the timezone pieces.

Since this is still experimental, we can delay the resolution of my timezone related comments by opening an issue to address that if it would block merging this by a considerable amount.

File: src/components/tz.rs
Comment (nekevss) [reply src/components/tz.rs:37]: I was sort of thinking along the similar lines, but figured I'd go with just for the sake of moving along. I can definitely switch it over.

File: src/tzdb.rs
Comment (nekevss) [reply src/tzdb.rs:98]: Yeah, I can add that method. I think my primary thoughts were that it would be nice to try and fix this in the `tzif` crate. But in the meantime, adding a method makes sense.

File: src/tzdb.rs
Comment (nekevss) [reply src/tzdb.rs:178]: Huh, I really like this solution, but it looks like it has a lifetime issue around `self.cache.get` having an immutable borrow and `self.cache.entry` borrowing it mutably.

File: src/tzdb.rs
Comment (nekevss) [reply src/tzdb.rs:133]: Yeah, if I remember correctly, I think there may have been a bug here, but it should also be solved by #100. If the value is past the transition times, then we need to use the POSIX tz string to resolve.

Ideally, at least the way I am understanding the behavior, we should return the first index if the `seconds` < transition_times[0]. if the value is greater than transition_times[n-1], then the POSIX tz string should be used to resolve the time zone.

File: src/tzdb.rs
Comment (jedel1043) [inline src/tzdb.rs:122]: My suggestion was more on the lines of removing `binary_search` and just doing this when doing the binary search:
```
let db = self.get_data_block2()?;
let result = db.transition_times.binary_search(epoch_seconds);

Comment (nekevss) [reply src/tzdb.rs:122]: Ooooooh, we definitely can. [Glibc](https://codebrowser.dev/glibc/glibc/time/tzfile.c.html#__tzfile_compute) does a somewhat gnarly combination of searching that I was intrigued in seeing if it was actually more optimized. Think I was being a bit overly optimistic in separating that out early :sweat_smile:

**Review by jedel1043 — APPROVED** [review]
Looks great!

**Pull request merged**
